cmd/oras/root/push.go

```diff
@@ -224,19 +233,40 @@ func runPush(ctx context.Context, opts pushOptions) error {
 	return opts.ExportManifest(ctx, memoryStore, root)
 }
 
-func updateDisplayOption(opts *oras.CopyGraphOptions, fetcher content.Fetcher, verbose bool) {
+func doPush(pack packFunc, copy copyFunc, dst oras.Target) (ocispec.Descriptor, error) {
+	if tracked, ok := dst.(*track.Target); ok {
+		defer tracked.Close()
+	}
+	// Push
+	return pushArtifact(dst, pack, copy)
+}
+
+func updateDisplayOption(opts *oras.CopyGraphOptions, fetcher content.Fetcher, verbose bool, tracked *track.Target) {
 	committed := &sync.Map{}
-	opts.PreCopy = display.StatusPrinter("Uploading", verbose)
 	opts.OnCopySkipped = func(ctx context.Context, desc ocispec.Descriptor) error {
 		committed.Store(desc.Digest.String(), desc.Annotations[ocispec.AnnotationTitle])
-		return display.PrintStatus(desc, "Exists   ", verbose)
+		return tracked.Prompt(desc, "Exists   ", verbose)
```

Comment:
This will panic if `tracked` is `nil`.

Reply:
No, it will not since `track.Target` explicitly handled `nil`

Reply:
Will change according to the suggestion in https://github.com/oras-project/oras/pull/1138#discussion_r1360530539